tests/transports/testquic.nim

```diff
@@ -424,3 +349,97 @@ suite "Quic transport":
     await runClient(server)
     await serverFut
     await server.stop()
+
+  asyncTest "multiaddress validation - accept valid addresses":
+    let transport = await createTransport()
+    defer:
+      await transport.stop()
```

Comment:
hint: `createTransport()` creates transport but does not start it. transport is started in `createTransport()` only if `isServer = true`. because transports used fro server sides do not need to be started.

https://github.com/vacp2p/nim-libp2p/pull/1784/files#diff-d96e6a34acc9f4de72802a718535853d536b6384bb901c3cbd6005575aa5cafdR79